### PR TITLE
fix(profiling): fix panic by explicitly depending on `ext-json`

### PR DESCRIPTION
**Pull request by realFlowControl**
### Description

This makes the profilers dependency on `ext-json` explicit, as well as fixes a bug on MacOS, where `php_json_serializable_ce` was `NULL` in every case (due to the missing `extern`) and the profiler would panic:
```
[2023-05-10T13:25:27.409327Z TRACE datadog_php_profiling] MINIT(1, 36)
thread '<unnamed>' panicked at 'assertion failed: tmp', src/config.rs:513:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
   2: core::panicking::panic
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:48:5
   3: datadog_php_profiling::config::minit
             at ./src/config.rs:513:9
   4: datadog_php_profiling::minit
             at ./src/lib.rs:211:5
   5: zend_startup_module_ex
             at /usr/local/src/php/php-82/Zend/zend_API.c:2235:7
   6: zend_startup_module_zval
             at /usr/local/src/php/php-82/Zend/zend_API.c:2250:10
   7: zend_hash_apply
             at /usr/local/src/php/php-82/Zend/zend_hash.c:2018:13
   8: zend_startup_modules
             at /usr/local/src/php/php-82/Zend/zend_API.c:2361:2
   9: php_module_startup
             at /usr/local/src/php/php-82/main/main.c:2260:2
  10: php_cli_startup
             at /usr/local/src/php/php-82/sapi/cli/php_cli.c:410:9
  11: main
             at /usr/local/src/php/php-82/sapi/cli/php_cli.c:1300:6
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fatal runtime error: failed to initiate panic, error 5
zsh: abort      RUST_BACKTRACE=1 php -d extension=target/debug/libdatadog_php_profiling.dylib
```

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [x] ~Tests added for this feature/bug.~

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: zend_abstract_interface/json/json.c [hunk 1]
```diff
@@ -22,7 +22,7 @@ __attribute__((weak)) int php_json_encode(smart_str *buf, zval *val, int options
 __attribute__((weak)) int php_json_decode_ex(zval *return_value, const char *str, size_t str_len, zend_long options,
                                              zend_long depth);
 #endif
-__attribute__((weak)) zend_class_entry *php_json_serializable_ce;
+extern __attribute__((weak)) zend_class_entry *php_json_serializable_ce;
```
Comment (morrisonlevi) [inline zend_abstract_interface/json/json.c:25]: Hmm. Using `extern` and `weak` on the same symbol doesn't make sense to me. What does that even mean?

File: profiling/src/lib.rs [hunk 2]
```diff
@@ -99,8 +99,10 @@ pub extern "C" fn get_module() -> &'static mut zend::ModuleEntry {
      * the result which avoids unsafe code and unnecessary locks.
      */
 
-    static DEPS: [zend::ModuleDep; 2] = [
+    static DEPS: [zend::ModuleDep; 4] = [
         // Safety: string is nul terminated with no interior nul bytes.
+        zend::ModuleDep::required(unsafe { CStr::from_bytes_with_nul_unchecked(b"standard\0") }),
+        zend::ModuleDep::required(unsafe { CStr::from_bytes_with_nul_unchecked(b"json\0") }),
```
Comment (morrisonlevi) [inline profiling/src/lib.rs:105]: In libdatadog's ddcommon, there is the `cstr!` macro which could be used here.

File: zend_abstract_interface/json/json.c (hunk 1 above)
Comment (realFlowControl) [reply zend_abstract_interface/json/json.c:27]: I am not 100% sure, but the following is happening on MacOS if `ext-json` is statically linked:
`php_json_serializable_ce` from `ext/json/json.c` will be initialised in the JSON extensions `MINIT`, but in our `zai_json_setup_bindings()` it is still `NULL`. Adding `extern` to it makes it show me the same pointer.

Comment (bwoebi) [reply zend_abstract_interface/json/json.c]: Macos actually requires extern weak... it doesn't make sense, but macos wants that.

Comment (realFlowControl) [reply zend_abstract_interface/json/json.c]: Just for future reference the Link Bob posted in Slack
https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/WeakLinking.html

File: profiling/src/lib.rs (hunk 2 above)
Comment (morrisonlevi) [reply profiling/src/lib.rs]: Yes, but we can remove the `// Safety` comment now.

Comment (realFlowControl) [reply profiling/src/lib.rs]: Done

File: zend_abstract_interface/json/json.h [hunk 3]
```diff
@@ -49,7 +49,11 @@ static inline int zai_json_decode_assoc(zval *return_value, const char *str, int
 }
 #endif
 
+#ifdef __APPLE__
+extern __attribute__((weak)) __attribute__((weak_import)) zend_class_entry *php_json_serializable_ce;
```
Comment (morrisonlevi) [inline zend_abstract_interface/json/json.h:53]: We should be able to use this shorter syntax, FYI:

```suggestion
extern __attribute__((weak, weak_import)) zend_class_entry *php_json_serializable_ce;
```

**Review by bwoebi — APPROVED** [review]

**Pull request merged**